src/Sarif/Core/PropertyBagHolder.cs

```diff
@@ -19,6 +19,12 @@ namespace Microsoft.CodeAnalysis.Sarif
     /// </summary>
     public class PropertyBagHolder : IPropertyBagHolder
     {
+        static PropertyBagHolder()
+        {
+            // Mitigation for Newtonsoft.Json v12 vulnerability GHSA-5crp-9r3c-p9vr
+            JsonConvert.DefaultSettings ??= () => new JsonSerializerSettings { MaxDepth = 64 };
```

Comment:
I still need to verify that this setting is, in fact, null in the default case.

Reply:
https://www.newtonsoft.com/json/help/html/P_Newtonsoft_Json_JsonSerializerSettings_MaxDepth.htm

Reply:
Good thought. The question is whether JsonConvert.DefaultSettings is null by default or not. The doc doesn't say.

Based on your help info, though, I can make a defensive code fix, thanks very much.

[https://www.newtonsoft.com/json/help/html/P_Newtonsoft_Json_JsonConvert_DefaultSettings.htm](https://www.newtonsoft.com/json/help/html/P_Newtonsoft_Json_JsonConvert_DefaultSettings.htm)

Reply:
OK, forgot this thing is a delegate. It *is*, in fact, null on start-up. My thought around initializing MaxDepth defensively was bogus, since the settings object is the output of a call to the delegate, when non-null. Anyway, I confirmed this code is correct.